include/chopper/pack/hierarchical_binning.hpp

```diff
@@ -38,22 +42,28 @@ class hierarchical_binning
     /*!\brief The constructor from user bin names, their kmer counts and a configuration.
      * \param[in, out] data_ The data input: filenames associated with the user bin and a kmer count per user bin.
      * \param[in] config_ A configuration object that holds information from the user that influence the computation.
+     * \param[in] ibf_model_ The model of the current level IBF for statistics generation
      * \param[in] query_cost The current cost to query all k-mers. Default: 0.0
      *
      * Each entry in the names_ and input vector respectively is considered a user bin (both vectors must have the
      * same length).
      */
-    hierarchical_binning(pack_data & data_, pack_config const & config_, double const query_cost = 0.0) :
+    hierarchical_binning(pack_data & data_, 
+                         pack_config const & config_, 
+                         hibf_model::ibf & ibf_model_,
+                         double const query_cost = 0.0) :
```

Comment:
Same here. It looks like the hierarchical binning algorithm depends on a model which is confusing.

Can the object live in `pack_data` instead?